### PR TITLE
[FLINK-23547][build] run integration tests on install

### PR DESCRIPTION
**Pull request by sjwiesman**
Stateful Functions is missing the surefire maven plugin and integration tests, marked with ITCase suffix, are not executed on install.

### CODE REVIEW

**Comment by sjwiesman** [comment]
Well, I guess the question is was this on purpose? I did find at least one existing ITCase for Unix sockets that never runs on CI

File: pom.xml [hunk 1]
```diff
@@ -307,6 +308,58 @@ under the License.
...
+            <!--surefire for unit tests and integration tests-->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.1</version>
+                <configuration>
+                    <trimStackTrace>false</trimStackTrace>
+                    <systemPropertyVariables>
+                        <forkNumber>0${surefire.forkNumber}</forkNumber>
+                        <checkpointing.randomization>true</checkpointing.randomization>
```
Comment (StephanEwen) [inline pom.xml:321]: This is Flink-specific, not needed in statefun.

File: pom.xml [hunk 2]
```diff
@@ -307,6 +308,58 @@ under the License.
...
+                        <!-- on, unset, or random -->
+                        <checkpointing.changelog>unset</checkpointing.changelog>
```
Comment (StephanEwen) [inline pom.xml:323]: Same as above

File: pom.xml [hunk 3]
```diff
@@ -307,6 +308,58 @@ under the License.
...
+                        <project.basedir>${project.basedir}</project.basedir>
+                        <!--suppress MavenModelInspection -->
+                        <test.randomization.seed>${test.randomization.seed}</test.randomization.seed>
```
Comment (StephanEwen) [inline pom.xml:326]: Same as above.

File: pom.xml [hunk 4]
```diff
@@ -307,6 +308,58 @@ under the License.
...
+                        <phase>test</phase>
+                        <goals>
+                            <goal>test</goal>
+                        </goals>
+                        <configuration>
+                            <includes>
+                                <include>${test.unit.pattern}</include>
+                            </includes>
+                        </configuration>
+                    </execution>
```
Comment (StephanEwen) [inline pom.xml:343]: Maybe add `<reuseForks>true</reuseForks>` here to make it explicit.

**Review by StephanEwen — COMMENTED** [review]
Looks good with some minor comments.

**Comment by sjwiesman** [comment]
Thanks for the review, will address when merging

**Pull request closed without merging**